miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/LibraryTemplateRestController.java

```diff
@@ -0,0 +1,192 @@
+    return Dtos.asDto(libraryTemplateService.get(id));
+  }
+  
+  @GetMapping(value = "/dt", produces = "application/json")
+  @ResponseBody
+  public DataTablesResponseDto<LibraryTemplateDto> getLibraryTemplates(HttpServletRequest request, HttpServletResponse response,
+      UriComponentsBuilder uriBuilder) throws IOException {
+    return jQueryBackend.get(request, response, uriBuilder);
+  }
+
+  @GetMapping(value = "/dt/project/{id}", produces = "application/json")
+  @ResponseBody
+  public DataTablesResponseDto<LibraryTemplateDto> getDTLibraryTemplatesByProject(@PathVariable("id") Long id, HttpServletRequest request,
+      HttpServletResponse response, UriComponentsBuilder uriBuilder) throws IOException {
+    return jQueryBackend.get(request, response, uriBuilder, PaginationFilter.project(id));
+  }
+
+  @PostMapping(value = "/bulk-delete")
+  @ResponseBody
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  public void bulkDelete(@RequestBody List<Long> ids) throws IOException {
+    List<LibraryTemplate> libraryTemplates = new ArrayList<>();
+    for (Long id : ids) {
+      if (id == null) {
+        throw new RestException("Cannot delete null LibraryTemplate", Status.BAD_REQUEST);
+      }
+      LibraryTemplate libraryTemplate = libraryTemplateService.get(id);
+      if (libraryTemplate == null) {
+        throw new RestException("LibraryTemplate " + id + " not found", Status.BAD_REQUEST);
+      }
+      libraryTemplates.add(libraryTemplate);
+    }
+    libraryTemplateService.bulkDelete(libraryTemplates);
+  }
+
+  @PostMapping(value = "/project/add")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  public void bulkAddProject(@RequestParam("projectId") Long projectId, @RequestBody List<Long> templateIds)
+      throws IOException {
+    List<LibraryTemplate> templates = templateIds.stream().map(WhineyFunction.rethrow(id -> libraryTemplateService.get(id)))
+        .collect(Collectors.toList());
+    templates.forEach(template -> {
+      try {
+        addProject(template, projectId);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    });
+  }
+
+  private void addProject(LibraryTemplate template, Long projectId) throws IOException {
+    List<Project> projects = template.getProjects();
+    projects.add(projectService.getProjectById(projectId));
+    template.setProjects(projects);
+    libraryTemplateService.update(template);
+  }
+
+  @PostMapping(value = "/project/remove")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  public void bulkRemoveProject(@RequestParam("projectId") Long projectId, @RequestBody List<Long> templateIds)
+      throws IOException {
+    List<LibraryTemplate> templates = templateIds.stream().map(WhineyFunction.rethrow(id -> libraryTemplateService.get(id)))
+        .collect(Collectors.toList());
+    templates.forEach(template -> {
+      try {
+        removeProject(template, projectId);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)